src/engraving/dom/edit.cpp

```diff
@@ -6519,7 +6526,13 @@ void Score::undoRemoveElement(EngravingItem* element, bool removeLinked)
             if (s->header() || s->trailer()) {        // probably more segment types (system header)
                 s->setEnabled(false);
             } else {
+                auto needAnchor = isSpannerStartEnd(s->tick(), element->track());
                 doUndoRemoveElement(s);
+                if (needAnchor) {
+                    TimeTickAnchor* anchor = EditTimeTickAnchors::createTimeTickAnchor(s->measure(), s->tick() - s->measure()->tick(), track2staff(element->track()));
+                    doUndoAddElement(anchor->parentItem());
+                    doUndoAddElement(anchor);
```

Comment:
code duplication here, if it's agreed that this is the correct fix, will clean up